src/charm.py

```diff
@@ -1942,6 +1969,22 @@ def push_tls_files_to_workload(self) -> bool:
         if cert is not None:
             self._push_file_to_workload(container, f"{self._storage_path}/{TLS_CERT_FILE}", cert)
 
+        key, ca, cert = self.tls.get_peer_tls_files()
+        if key is not None:
+            self._push_file_to_workload(
+                container, f"{self._storage_path}/peer_{TLS_KEY_FILE}", key
+            )
+        if ca is not None:
+            self._push_file_to_workload(container, f"{self._storage_path}/peer_{TLS_CA_FILE}", ca)
+        if cert is not None:
+            self._push_file_to_workload(
+                container, f"{self._storage_path}/peer_{TLS_CERT_FILE}", cert
+            )
+
+        # CA bundle is not secret
+        with open(f"/tmp/{TLS_CA_BUNDLE_FILE}", "w") as fp:  # noqa: S108
+            fp.write(self.tls.get_peer_ca_bundle())
```

Comment:
The CAs bundle is needed by the charm code to make sure it can verify REST API calls, so it is only pushed on the `charm` container.